addons/event_booth_sale/models/sale_order_line.py

```diff
@@ -25,13 +26,20 @@ def _compute_name_short(self):
             record.name_short = record.event_booth_pending_ids.event_id.name
         super(SaleOrderLine, self - wbooth)._compute_name_short()
 
+    @api.constrains('event_booth_registration_ids')
+    def _check_event_booth_registration_ids(self):
+        if len(self.event_booth_registration_ids.mapped('event_booth_id.event_id')) > 1:
+            raise ValidationError(_('Registrations from the same Order Line must belong to a single event.'))
+
     @api.onchange('product_id')
-    def _onchange_booth_product_id(self):
+    def _onchange_product_id(self):
+        """We reset the event when the selected product doesn't belong to any pending booths."""
         if self.event_id and (not self.product_id or self.product_id not in self.event_booth_pending_ids.product_id):
             self.event_id = None
 
     @api.onchange('event_id')
-    def _onchange_booth_event_id(self):
+    def _onchange_event_id(self):
```

Comment:
Just ot be sure, could be _onchange_event_id_booth (does not seem to have one in event_sale, but hey).